onnxruntime/core/providers/cpu/reduction/reduction_ops.cc

```diff
@@ -137,7 +137,20 @@ bool PrepareForReduce(OpKernelContext* ctx,
   ORT_ENFORCE(input_tensor_ptr != nullptr);
   const Tensor& input = *input_tensor_ptr;
 
-  size_t ndim = input.Shape().GetDims().size();
+  size_t ndim = input.Shape().NumDimensions();
+
+  // Scalar tensor
+  if (ndim == 0) {
```

Comment:
Would it be simpler to just check input.Shape().Size() == 1? that handles both forms of scalar (rank 0 and rank 1 with dim value of 1).

Reply:
The issue here is that the shape may be (1, 1, 1, 1....) and I decided not to mess with dimension reduction and let the original code handle it.

---
In reply to: [394737729](https://github.com/microsoft/onnxruntime/pull/3260#discussion_r394737729) [](ancestors = 394737729)